src/parser/mod.rs

```diff
@@ -11833,7 +11847,21 @@ impl<'a> Parser<'a> {
             };
             joins.push(join);
         }
-        Ok(TableWithJoins { relation, joins })
+        Ok(joins)
+    }
+
+    fn is_next_token_a_join(&self) -> bool {
```

Comment:
```suggestion
    fn peek_parens_less_nested_join(&self) -> bool {
```
maybe something like this naming wise? peek uses existing parser terminology. Also would it be possible with a doc comment of the function mentioning the use case? I think that would be helpful to future readers